src/app/EventManagement.h

```diff
@@ -298,6 +289,25 @@ class EventManagement
 
     static void DestroyEventManagement();
 
+    class LockEvents
+    {
+    public:
+        LockEvents(EventManagement & aEventManagement) : mEventManagement(aEventManagement)
+        {
+#if !CHIP_SYSTEM_CONFIG_NO_LOCKING
```

Comment:
This whole CHIP_SYSTEM_CONFIG_NO_LOCKING define seems odd here. You cannot have something called a Lock in code that you can't tell unless you look at the implementation that it does nothing. Which platforms have this define and why? Could you call it ProtectEvents instead? Where are the side effects of CHIP_SYSTEM_CONFIG_NO_LOCKING being used called out?
.

Reply:
just rename it to ScopedLock, we have enable lock for all supported platform, so it is safe to remove CHIP_SYSTEM_CONFIG_NO_LOCKING here.